lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,429 @@
+      action: lambda:InvokeFunction
+      principal: s3.amazonaws.com
+      source_arn: arn:aws:s3:eu-central-1:123456789012:bucketName
+      source_account: 123456789012
+
+  - name: show results
+    debug: var=lambda_policy_action
+
+'''
+
+RETURN = '''
+---
+lambda_policy_action:
+    description: describes what action was taken
+    returned: success
+    type: string
+'''
+
+import json
+import re
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import get_aws_connection_info, boto3_conn
+
+try:
+    from botocore.exceptions import ClientError
+except:
+    pass  # will be protected by AnsibleAWSModule
+
+
+def pc(key):
+    """
+    Changes python key into Pascal case equivalent. For example, 'this_function_name' becomes 'ThisFunctionName'.
+
+    :param key:
+    :return:
+    """
+
+    return "".join([token.capitalize() for token in key.split('_')])
+
+
+def policy_equal(module, current_statement):
+    for param in ('action', 'principal', 'source_arn', 'source_account', 'event_source_token'):
+        if module.params.get(param) != current_statement.get(param):
+            return False
+
+    return True
+
+
+def set_api_params(module, module_params):
+    """
+    Sets module parameters to those expected by the boto3 API.
+
+    :param module:
+    :param module_params:
+    :return:
+    """
+
+    api_params = dict()
+
+    for param in module_params:
+        module_param = module.params.get(param)
+        if module_param is not None:
+            api_params[pc(param)] = module_param
+
+    return api_params
+
+
+def validate_params(module):
+    """
+    Performs parameter validation beyond the module framework's validation.
+
+    :param module:
+    :return:
+    """
+
+    function_name = module.params['function_name']
+
+    # validate function name
+    if function_name.startswith('arn:'):
+        if not re.search('^[\w\-]+$', function_name):
+            module.fail_json(
+                msg='Function name {0} is invalid. Names must contain only alphanumeric characters and hyphens.'.format(
+                    function_name)
+            )
+        if len(function_name) > 64:
+            module.fail_json(
+                msg='Function name "{0}" exceeds 64 character limit'.format(function_name))
+    else:
+        if not re.search('^[\w\-:]+$', function_name):
+            module.fail_json(
+                msg='ARN {0} is invalid. ARNs must contain only alphanumeric characters, hyphens and colons.'.format(function_name)
+            )
+        if len(function_name) > 140:
+            module.fail_json(msg='ARN name "{0}" exceeds 140 character limit'.format(function_name))
+
+
+def get_qualifier(module):
+    """
+    Returns the function qualifier as a version or alias or None.
+
+    :param module:
+    :return:
+    """
+
+    if module.params.get('version') is not None:
+        return str(module.params['version'])
```

Comment:
Should this be using to_text() here and on line 218?

Reply:
Yes. Either to_native or to_text would be fine in this case.